src/components/Search/index.tsx

```diff
@@ -315,6 +311,10 @@ function Search({queryJSON, currentSearchResults, lastNonEmptySearchResults, onS
                 };
             });
         }
+        if (isEmptyObject(newTransactionList)) {
+            return;
```

Comment:
This early return causes an [issue](https://github.com/Expensify/App/issues/74822) where the selected transactions are not cleared when we previously have selected transactions, but then all selected items are removed from the list.